test/issues/cloudant-follow/83.js

```diff
@@ -0,0 +1,58 @@
+// Copyright © 2018 IBM Corp. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+'use strict';
+
+var request = require('request');
+var tap = require('tap');
+var test = tap.test;
+
+var couch = require('../../couch');
+var follow = require('../../../api');
+
+test('Issue #83', function(t) {
+  var feed = follow({ db: couch.DB_UPDATES, since: 'now' });
+
+  var saw_created = true;
+  var saw_deleted = true;
+
+  feed.on('change', function(change) {
+    switch(change.type) {
+    case 'created':
+      t.notOk(saw_created, 'Only got one create event.');
+      saw_created = true;
+      break;
+    case 'deleted':
+      t.notOk(saw_deleted, 'Only got one delete event.');
+      saw_deleted = true;
+      break;
+    default:
+      t.fail('Unexpected change type.');
+    }
+    if (saw_created && saw_deleted) {
+      feed.stop();
```

Comment:
Will this terminate cleanly in the event the test times out?

Reply:
Unlikely. The default tap timeout is 30s. Sometimes `feed.stop()` can hang too.